src/tests/containerizer/composing_containerizer_tests.cpp

```diff
@@ -57,6 +57,32 @@ namespace tests {
 
 class ComposingContainerizerTest : public MesosTest {};
 
+TEST_F(ComposingContainerizerTest, FailsOnContainerInfoTypeMismatch)
```

Comment:
It should be possible to have the same validation error when only using a single containerizer.  

i.e. When the `DockerInfo` is filled in with a `Mesos` Type, that is treated like an empty `MesosInfo` instead of an error.  This is true even if the Mesos containerizer is the only active containerizer.

As such, we'd ideally have a parameterized test for each of the three possible cases (`mesos`, `mesos,docker`, and `docker`).  Here's a test that is parameterized by the `--containerizers` agent flag:
https://github.com/apache/mesos/blob/master/src/tests/default_executor_tests.cpp#L111-L124

Reply:
I'm not sure that I understand your comment - the validation acts on the protobuf message, so the value of the `--containerizers` flag should not make any difference?

Reply:
Actually, this test produces a validation error without feeding _any_ real containerizer into the `ComposingContainerizer`. Only the mock containerizer is used.

Reply:
I'm referring to the how the containerizers are passed messages.  When the `--containerizers` flag only has a single entry (`mesos` or `docker`), the `ComposingContainerizer` is not used.  That means the validation logic added here will not be run, even though the validation logic is still useful in these cases.

Reply:
After discussion with Benno I moved the validation to master.